Lib/inspect.py

```diff
@@ -2028,7 +2028,9 @@ def _signature_get_partial(wrapped_sig, partial, extra_args=()):
             if param.kind is _POSITIONAL_ONLY:
                 # If positional-only parameter is bound by partial,
                 # it effectively disappears from the signature
-                new_params.pop(param_name)
+                # However, if it is a Placeholder it is not removed
+                if arg_value != functools.Placeholder:
```

Comment:
Use `is`. `arg_value` can have custom `__eq__` which always return True. For example, `unittest.mock.ANY`. BTW, please add a test with `unittest.mock.ANY`.